src/Encoding/MicrosecondBasedDateConversion.php

```diff
@@ -25,16 +25,16 @@ public function formatClaims(array $claims): array
         return $claims;
     }
 
-    /** @return int|string */
+    /** @return int|float */
     private function convertDate(DateTimeImmutable $date)
     {
-        $seconds      = $date->format('U');
+        $timestamp    = $date->format('U.u');
         $microseconds = $date->format('u');
 
         if ((int) $microseconds === 0) {
-            return (int) $seconds;
+            return (int) $timestamp;
         }
 
-        return $seconds . '.' . $microseconds;
+        return (float) $timestamp;
```

Comment:
I believe this whole function can be replaced with `return (float)$date->format('U.u');`

Reply:
That's correct, it simplifies everything.